lib/actions.ts

```diff
@@ -2,15 +2,39 @@
 
 import * as db from "@/lib/db";
 import * as Schemas from "drizzle/schema";
+import { signIn } from "@/lib/auth";
 import { auth } from "./auth";
 import * as blob from "@vercel/blob";
-import { NewProjectFormSchema } from "./types";
+import { NewProjectFormSchema, RegisterSchemaType } from "./types";
 import { redirect } from "next/navigation";
 import { defaulTaskSpecs } from "./tasks";
 import assert from "assert";
 
 const VERCEL_BLOB_FAKE_FILES = true;
 
+export async function registerUser(data: RegisterSchemaType) {
+  try {
+    let user = await db.getUser(data.email);
```

Comment:
Another thing to consider is synchronicity
it's possible for two people to try and register with the same email at the same time. In this case two registerUser functions would be run at the same time (I assume this is how the nextjs backend works) and both of those function instances would return no users here and both would call createuser.

However , this is not such a big deal as the email schema labels email as unique, and database edits like this happen one by one, and not in parallel.
https://github.com/patrickchin/haru2-vercel/blob/manage/drizzle/schema.ts#L22

It is good practice to do things "atomically", using a transaction. I recently added one here on this branch:
https://github.com/patrickchin/haru2-vercel/blob/manage/lib/db.ts#L157
In this case, i can't delete the team until all team members are deleted. But to make sure no team members are added between deleting all the members and deleting the team, they have to be done in a transaction so no other changes can be made to the database in the meantime. Or if there are, you won't get partial edits of the database and it will roll back all of the changes.

TBH this is a whole lot of yapping, and I just realised you don't actually need to do a transaction or even do this check, the .unique() already does that check for you